scalameta/contrib/src/main/scala/scala/meta/contrib/implicits/Statements.scala

```diff
@@ -0,0 +1,105 @@
+package scala.meta.contrib.implicits
+
+import scala.collection.immutable.Seq
+import scala.language.higherKinds
+import scala.language.implicitConversions
+import scala.meta._
+import scala.meta.contrib.Equal
+
+/**
+  * Typeclass to deal with scalameta trees which have an expected set of statements.
+  *
+  * For trees that have multiple stat lists (such as Defn.Class) the main stats are used.
+  *
+  * This allows you to do things like add a statement to the
+  * body of a class without digging through the low level meta api
+  *
+  * TODO: Consider using a meta variant of simulacrum to
```

Comment:
Could you explain the value in organizing type-classes the simulacrum way? Compared to 

```scala
trait TypeClass[T] {
  def op(e: T): Foo
}
object TypeClass {
  implicit val XTypeClass: TypeClass[X]
  ...
}

package object contrib {
  implicit class XtensionTypeClass[A](e: A)(implicit ev: TypeClass[A]) {
    def op: Foo = ???
    ...
  }
}}}
}
```